### PR TITLE
ARC-1362: add security FAQ

### PR DESCRIPTION
**Pull request by thombergs**
Adds a security FAQ to the docs.

Use this link for a nicer view: https://github.com/atlassian/github-for-jira/blob/ARC-1362-security-faq/docs/security.md

### CODE REVIEW

**Review by mboudreau — APPROVED** [review]

File: docs/security.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+| Read-only access to [members](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#permission-on-members)                                                                                                                                   | The needs read access to the members of your GitHub organizations to determine whether they are an admin of the org or not. Only admins are allowed to change the configuration of the GitHub for Jira app.                                                                                                                                   |
+
+The GitHub for Jira app has `READ`, `WRITE`, and `DELETE` permissions on the Jira API. The permissions to the Jira API cannot be finer grained, at the moment, so that these permissions are scoped across the whole Jira API.
```
Comment (bgvozdev) [inline docs/security.md:53]: I was wondering if we should use "The" when we are referring to the app by name (here and in other places)...

File: docs/security.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+The database and the compute servers of the GitHub for Jira app are located in AWS’s us-west region.
+
+**GitHub Cloud + GitHub Server:**
+
+| Field | Description                                                                                                                                                  |
+| ----- |--------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Jira host | The hostname of your Jira site (i.e. *.atlassian.net). This is used as an identifier for your Jira site.                                                     
+| Connect client key | The client key that is generated during installation of the app into your Jira site. This key is used to authorize all calls to your Jira site’s API.        |
+| Connect shared secret | The client secret that is generated during installation of the app into your Jira site. This secret is used to authorize all calls to your Jira site’s API.  |
```
Comment (bgvozdev) [inline docs/security.md:128]: I think it is also encrypted with cryptor, right @gxueatlassian ?

File: docs/security.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+> 
+> If you use the GitHub for Jira app with a GitHub Enterprise Server instance that you control yourself, you are prompted to create a GitHub app on that server. This GitHub app has a private key that will be uploaded to the GitHub for Jira app and stored in its database (see “GitHub app private key” in the table above).
+> 
+> At first glance it may seem inappropriate to share a private key because it is private, after all. However, this key enables the GitHub for Jira app to sign JWTs that are used to access your GitHub organization’s API so GitHub can make sure that the requests are coming from a valid source.
+> 
+> Since you are creating the GitHub app specifically for the GitHub for Jira integration, you can revoke access at any time by removing the private key from your GitHub server.
+
+## How can I delete my data?
+
+All data concerning your GitHub organization that is stored by the GitHub for Jira app’s database is deleted when you uninstall the GitHub for Jira app from your Jira site.
```
Comment (bgvozdev) [inline docs/security.md:163]: Maybe "metadata"? The data itself is proxied to Jira API and is not stored..

File: docs/security.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+## How is the communication secured?
+
+All HTTP traffic is secured with TLS (HTTPS).
```
Comment (bgvozdev) [inline docs/security.md:167]: If GHE is not using https, we will use http...

File: docs/security.md (hunk 1 above)
Comment (thombergs) [reply docs/security.md:53]: Yeah, I think the "the" is appropriate

File: docs/security.md (hunk 2 above)
Comment (thombergs) [reply docs/security.md:128]: yes, it is, I looked it up in the code

**Review by bgvozdev — APPROVED** [review]

**Pull request merged**